core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -513,13 +516,44 @@ private[spark] object Utils extends Logging {
       Files.move(sourceFile, destFile)
```

Comment:
Tested this in isolation and it does work for directories.  A new test that I added should exercise this path.